src/pages/index.js

```diff
@@ -24,40 +32,141 @@ const profileEditModalClass = new ModalWithForm(
   "#profile-edit-modal",
   handleProfileFormSubmit
 );
+const profileEditAvatarModalClass = new ModalWithForm(
+  "#profile-edit-avatar-modal",
+  handleProfileAvatar
+);
 
+const confirmationModalClass = new ModalWithConfirmation("#confirmation-modal");
 const previewImageModalClass = new ModalWithImage("#preview_image_modal");
 
-// Functions
+// GET API
+
+api
+  .getUserInfo()
+  .then((data) => {
+    userInfoClass.setUserInfo({ title: data.name, description: data.about });
+    userInfoClass.setUserAvatar({
+      avatar: data.avatar,
+    });
+    return data;
+  })
+  .catch((error) => {
+    console.error("Error fetching user info:", error);
+  });
 
-const sectionClass = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
-);
-sectionClass.renderItems();
+let sectionClass;
+
+api
+  .getInitialCards()
+  .then((cardList) => {
+    sectionClass = new Section(
+      { items: cardList, renderer: createCard },
+      ".cards__list"
+    );
+    sectionClass.renderItems();
+  })
+  .catch((error) => {
+    console.error("Error fetching user info:", error);
+  });
 
+// PATCH API
 function handleProfileFormSubmit(formData) {
-  userInfoClass.setUserInfo(formData);
+  profileEditModalClass.setLoading(true, "Save");

```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```